src/main/kotlin/org/opensearch/commons/alerting/AlertingPluginInterface.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.commons.alerting
+
+import org.opensearch.action.ActionListener
+import org.opensearch.action.ActionResponse
+import org.opensearch.client.node.NodeClient
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.commons.alerting.action.AlertingActions
+import org.opensearch.commons.alerting.action.IndexMonitorRequest
+import org.opensearch.commons.alerting.action.IndexMonitorResponse
+import org.opensearch.commons.utils.BaseResponse
+import org.opensearch.commons.utils.recreateObject
+
+object AlertingPluginInterface {
+
+    fun indexMonitor(
+        client: NodeClient,
+        request: IndexMonitorRequest,
+        listener: ActionListener<IndexMonitorResponse>
+    ) {
+        client.execute(
+            AlertingActions.INDEX_MONITOR_ACTION_TYPE,
+            request,
+            wrapActionListener(listener) { response ->
+                recreateObject(response) {
+                    IndexMonitorResponse(
+                        it
+                    )
+                }
+            }
+        )
+    }
+
+    @Suppress("UNCHECKED_CAST")
```

Comment:
lets remove uncheck casting

Reply:
this check suppresses 2 warnings. So, will keep it as `Alerting` to `Notifications` follow similar pattern.